projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+// Comments ----------
+// I write a lot of comments, it has helped me the entire course to learn, understand and remember
+// They can all be removed at the end to clean it up
+
+func handleOKResponse(response *http.Response) {
+	// try to read the response body
+	responseBody, err := io.ReadAll(response.Body)
+
+	// use the defer keyword early to close the body regardless
+	defer response.Body.Close()
+
+	if err != nil {
```

Comment:
In go, best practice is to check for errors right after they are given value. So I'd suggest to move `defer response.Body.Close()` either before you read from body or after you check for errors (preferred)

There's nothing technically wrong with this code, but can you think of why we would need to check for errors before anything else?